frontend/packages/dev-console/src/components/health-checks/AddHealthChecks.tsx

```diff
@@ -53,7 +54,9 @@ const AddHealthChecks: React.FC<FormikProps<FormikValues> & AddHealthChecksProps
     setFieldValue('containerName', containerName);
     setFieldValue('healthChecks', getHealthChecksData(resource, containerIndex));
     history.replace(
-      `/k8s/ns/${resource.metadata.namespace}/${resource.kind}/${resource.metadata.name}/containers/${containerName}/health-checks`,
+      `/k8s/ns/${resource.metadata.namespace}/${
+        modelFor(kindForCRDResource).crd ? kindForCRDResource : resource.kind
+      }/${resource.metadata.name}/containers/${containerName}/health-checks`,
```

Comment:
Not sure why this didnt came up before but the URL here is huge and ugly. You might want to spread `resource` somewhere above and then use the respective props. Also you can remove the crd check from the URL itself:

```
const kind = modelFor(kindForCRDResource).crd ? referenceFor(resource) : resource.kind
```
```suggestion
      `/k8s/ns/${namespace}/${kind}/${name}/containers/${containerName}/health-checks`,
```